feat: add gradle-use plugin (#1559)
canonical/craft-parts: docs/common/craft-parts/reference/plugins/gradle_use_plugin.rst

.. _craft_parts_gradle_use_plugin:

Gradle Use plugin
=================

The Gradle Use plugin builds Java projects using the Gradle build tool and, unlike
the :ref:`craft_parts_gradle_plugin`, publishes the artifacts to a local `Maven
repository`_.

Keys
----

This plugin provides the following unique keys.

gradle-init-script
~~~~~~~~~~~~~~~~~~

**Type:** string

The path to the initialization script to run with ``gradle --init-script
<gradle-init-script>`` command. See `official gradle documentation
<https://docs.gradle.org/current/userguide/init_scripts.html>`_ on the init script.

gradle-parameters
~~~~~~~~~~~~~~~~~

**Type:** list of strings

Used to add additional parameters to the ``gradle publish`` command line.

gradle-use-daemon
~~~~~~~~~~~~~~~~~

> 
> **Type:** boolean
> 
> Whether to use the `Gradle daemon <https://docs.gradle.org/current/userguide/gradle_daemon.html>`_
> during the build. The daemon is disabled by default.

Reviewer: ```suggestion
**Type:** boolean
**Default:** false

Whether to use the `Gradle daemon <https://docs.gradle.org/current/userguide/gradle_daemon.html>`_
during the build.
```
